internal/traffic_shaping.py

```diff
@@ -560,6 +560,16 @@ def install(self):
                     else:
                         subprocess.call(['sudo', 'modprobe', 'ifb'])
                     subprocess.call(['sudo', 'ip', 'link', 'set', 'dev', 'ifb0', 'up'])
+                    subprocess.call(['sudo', 'tc', 'filter', 'add', 'dev', self.interface, 'parent',
+                                     'ffff:', 'protocol', 'ip', 'prio', '1',
+                                     'u32', 'match', 'ip', 'protocol', '17', '0xff',
+                                     'match', 'ip', 'dport', '514', '0xffff',
+                                     'action', 'pass'])
+                    subprocess.call(['sudo', 'tc', 'filter', 'add', 'dev', self.interface, 'parent',
+                                     'ffff:', 'protocol', 'ip', 'prio', '2',
+                                     'u32', 'match', 'ip', 'protocol', '6', '0xff',
+                                     'match', 'ip', 'dport', '22', '0xffff',
+                                     'action', 'pass'])
```

Comment:
Can you test the netem settings on a test machine stand-alone with something like iperf to make sure the 514 UDP and 22 TCP traffic is not shaped while 53 UDP and 80/443 TCP are shaped?

I don't know how well the filters here for "pass" work with the qdisc's added later when shaping is enabled.

Reply:
Also make sure the port 22 shaping is bypassed in both directions.

Reply:
> I don't know how well the filters here for "pass" work with the qdisc's added later when shaping is enabled.

You are correct. These are out of order. Changing.

Reply:
FWIW, it might be cleaner/easier to apply the filter to both interfaces, independent of protocol and both for src and dst (test traffic will never be from/to those ports).

Something like:

```python
for interface in [self.interface, 'ifb0']:
	for port in ['22', '514']:
		for direction in ['src', 'dst']:
			<tc filter command with direction, interface and port>
```

That will get you ~8 filter commands but it will make it easier to add ports later if needed and keep the code clean (once you figure out how to get the ordering/priorities working well).

Reply:
@pmeenan I need 514 to be `udp` only (protocol 17) so I'd need to do this in two separate loops, which is fine, but also is there much of a point at this time?

Reply:
`need`?  Technically syslog can run over TCP even though the agent will only be configured for UDP. There will never be any test browser traffic on that port so bypassing unilaterally for TCP and UDP would be fine.

Reply:
oh, fair. didn't really think about that point. just need to make sure it's working with udp ... doesn't matter how else it works. ok.

Reply:
seems like

```python
                    for interface in [self.interface, 'ifb0']:
                        for port in ['22', '514']:
                            for direction in ['dport', 'sport']:
                                subprocess.call(['sudo', 'tc', 'filter', 'add', 'dev', interface, 'parent',
                                                 'ffff:', 'protocol', 'ip', 'prio', '1',
                                                 'u32', 'match', 'ip', direction, port, '0xffff',
                                                 'flowid', '1:1', 'action', 'pass'])
```
will get us there.